usaspending_api/awards/v2/data_layer/orm.py

```diff
@@ -271,6 +274,15 @@ def fetch_award_details(filter_q, mapper_fields):
     return Award.objects.filter(**filter_q).values(*vals).annotate(**ann).first()
 
 
+def fetch_parent_award_from_piid(piid):
```

Comment:
There should be a better way to find the correct parent. `piid` isn't unique, and the generated_unique_award_id string can contain two piids.

I think both `parent_award_piid` and `fpds_parent_agency_id` is necessary when looking for a parent.
```python
Award.objects.filter(piid=<parent_award_piid>, fpds_agency_id=<fpds_parent_agency_id>).values(...).first()
```
@kbard for confirmation

Reply:
>  I think both parent_award_piid and fpds_parent_agency_id is necessary when looking for a parent.

I believe this to be a true and correct statement.  It can be seen in the `unique_award_key` which contains both the transaction's `piid` and `agency_id` and its parent's `piid` and `agency_id` (if it has a parent).